src/node/chainstatemanager_notifications.h

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_NODE_CHAINSTATEMANAGER_NOTIFICATIONS_H
+#define BITCOIN_NODE_CHAINSTATEMANAGER_NOTIFICATIONS_H
+
+#include <kernel/chainstatemanager_opts.h>
+
+namespace node {
+kernel::ChainstateManagerNotificationCallbacks DefaultChainstateManagerNotifications();
```

Comment:
I don't think I would tie these notifications to the ChainStateManager class specifically as they really make sense anywhere in validation code. Would suggest changing `chainstatemanager_notifications` to  `chainstate_notifications`, or `validation_notifications` and dropping the "manager" in these places

Reply:
Would you also prefer passing this `validation_notifications` class to the other users of these notification functions, even if they only use one of the methods?

Reply:
Yes, I think that's preferable. Just pass `const ChainstateNotifications&` everywhere and don't have a lot of different awkward notifications